lnbits/core/crud.py

```diff
@@ -339,37 +339,14 @@ async def delete_expired_invoices(
           AND time < {db.timestamp_now} - {db.interval_seconds(2592000)}
         """
     )
-
-    # then we delete all expired invoices, checking one by one
-    rows = await (conn or db).fetchall(
+    # then we delete all invoices whose expiry date is in the past
+    await (conn or db).execute(
         f"""
-        SELECT bolt11
-        FROM apipayments
-        WHERE pending = true
-          AND bolt11 IS NOT NULL
-          AND amount > 0 AND time < {db.timestamp_now} - {db.interval_seconds(86400)}
+        DELETE FROM apipayments
+        WHERE pending = true AND amount > 0
+          AND expiry < {db.timestamp_now}
```

Comment:
Delete task operates on database column `expiry` instead of looping through all invoices and calculating their expiry date one at a time.

Reply:
why the amount required here in that query?

Reply:
So we look at only invoices (incoming) and not payments (outgoing).